huaweicloud/services/ddm/resource_huaweicloud_ddm_instance.go

```diff
@@ -0,0 +1,931 @@
+	}
+	updateInstanceFlavorOpt.JSONBody = utils.RemoveNil(buildUpdateInstanceFlavorBodyParams(d, specCode))
+	_, err = updateInstanceFlavorClient.Request("PUT", updateInstanceFlavorPath, &updateInstanceFlavorOpt)
+	if err != nil {
+		return diag.Errorf("error updating DdmInstance: %s", err)
+	}
+
+	stateConf := &resource.StateChangeConf{
+		Pending:      []string{"RESIZING"},
+		Target:       []string{"RUNNING"},
+		Refresh:      ddmInstanceStatusRefreshFunc(d.Id(), region, cfg),
+		Timeout:      d.Timeout(schema.TimeoutUpdate),
+		Delay:        100 * time.Second,
+		PollInterval: 10 * time.Second,
+	}
+
+	_, err = stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return diag.Errorf("error waiting for instance (%s) to running: %s", d.Id(), err)
+	}
+	return nil
+}
+
+func getSpecCodeByFlavorId(d *schema.ResourceData, client *golangsdk.ServiceClient, flavorId, engineId string) (string, diag.Diagnostics) {
+	// getDdmFlavors: Query the List of DDM flavors
+	var (
+		getDdmFlavorsHttpUrl = "v2/{project_id}/flavors"
+	)
+
+	getDdmFlavorsPath := client.Endpoint + getDdmFlavorsHttpUrl
+	getDdmFlavorsPath = strings.ReplaceAll(getDdmFlavorsPath, "{project_id}", client.ProjectID)
+
+	getDdmFlavorsQueryParams := buildGetFlavorsQueryParams(engineId, 0)
+	getDdmFlavorsPath += getDdmFlavorsQueryParams
+	getInstanceOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+	}
+
+	for {
+		getDdmFlavorsResp, err := client.Request("GET", getDdmFlavorsPath, &getInstanceOpt)
+		if err != nil {
+			return "", common.CheckDeletedDiag(d, err, "error retrieving DdmFlavors")
+		}
+		getDdmFlavorsRespBody, err := utils.FlattenResponse(getDdmFlavorsResp)
+		if err != nil {
+			return "", diag.FromErr(err)
+		}
+		specCode, pageRes := flattenGetFlavorsResponseBody(getDdmFlavorsRespBody, flavorId)
+		if specCode != "" {
+			return specCode, nil
+		}
+		if pageRes.offset+pageRes.limit >= pageRes.x86Total && pageRes.offset+pageRes.limit >= pageRes.armTotal {
+			break
+		}
+		getDdmFlavorsPath = updatePathOffset(getDdmFlavorsPath, pageRes.offset+pageRes.limit)
+	}
+	return "", diag.Errorf("can not found flavor by flavorId: %s", flavorId)
+}
+
+func flattenGetFlavorsResponseBody(resp interface{}, flavorId string) (string, *queryRes) {
+	if resp == nil {
+		return "", &queryRes{}
+	}
+	curJson := utils.PathSearch("computeFlavorGroups", resp, make([]interface{}, 0))
+	curArray := curJson.([]interface{})
+
+	var offset, limit, x86Total, armTotal float64
+	var specCode string
+	for _, v := range curArray {
+		specCode = flattenFlavors(v, flavorId)
+		if specCode != "" {
+			return specCode, &queryRes{}
+		}
+		offset = utils.PathSearch("offset", v, 0).(float64)
+		limit = utils.PathSearch("limit", v, 0).(float64)
+		flavorCPUArch := utils.PathSearch("groupType", v, nil)
+		if flavorCPUArch == "X86" {
+			x86Total = utils.PathSearch("total", v, 0).(float64)
+		} else {
+			armTotal = utils.PathSearch("total", v, 0).(float64)
```

Comment:
It seems like `x86Total` and `armTotal` use the same function. 
If can make one variable， and remove this if ... else...

Reply:
If curArray length greater than 1，local variable may be covered.

Reply:
the response will contain x86 data and arm data,  then x86Total and armTotal are both needed, but the result limit and offset is same, so just keep one